Add initial AAT specification with error codes (#56)
ably/ably-asset-tracking-common: specification/README.md

# Ably Asset Tracking Specification

## Overview

> This document is meant to be a formal, canonical specification describing the Ably Asset Tracking solution.

Reviewer: In line with the wider description in #45, which we've discussed outside scope of this PR but have agreed this PR represents the starting point for, I think we can refine this paragraph and make it a little bit more assertive:

```suggestion
This document is the formal, canonical specification describing the ecosystem, including requirements, shared by components of Ably's [Asset Tracking solution](https://ably.com/solutions/asset-tracking) (AAT).
```

Author: https://github.com/ably/ably-asset-tracking-common/pull/56/commits/77941d29ac2675a30b08a4e76157d9d38349005b

## Error codes

The list of Ably Asset Tracking specific error codes.

> | Code   | Name            | Description                                                                                                                                    |

Reviewer: We can never predict how wide any of the columns will every need to be in markdown tables, so I always compact the heading rows down (more Git diff friendly as the table evolves)...

```suggestion
| Code | Name | Description |
```

Author: The auto-formatting did that for me and I had a feeling that it is not particularly good :sweat_smile: https://github.com/ably/ably-asset-tracking-common/pull/56/commits/bf751cb491e50ddbe68e18bd43b8da9a4b9a1c31

> | ------ | --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |

Reviewer: ...and...

```suggestion
| ---- | ---- | ----------- |
```

to match. 😁

Author: https://github.com/ably/ably-asset-tracking-common/pull/56/commits/bf751cb491e50ddbe68e18bd43b8da9a4b9a1c31
